src/backend/parser/parse_func.c

```diff
@@ -2063,6 +2063,10 @@ funcname_signature_string(const char *funcname, int nargs,
 
 	initStringInfo(&argbuf);
 
+	if (sql_dialect == SQL_DIALECT_TSQL && remove_db_name_in_schema_hook) {
```

Comment:
would it be better to put this hook in `func_signature_string()` we already have the splitted name list there.
Also dialect check inside hook function.